src/convertToObject.js

```diff
@@ -3,16 +3,53 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ *  (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ *  (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const cssStyles = sourceString
+    .split(';')
+    .map(str => {
+      return str
+        .split('\n')
+        .filter(item => item !== '')
+        .map((item, i) => {
+          return item
+            .split(' ')
+            .filter(el => el !== '')
+            .map(el => {
+              return el
+                .split(':')
+                .filter(elWithoutEmptyString => elWithoutEmptyString !== '');
+            })
+            .filter(el => el.length !== 0)
+            .map(el => el[0]);
+        })
+        .filter(el => el.length !== 0);
+    })
+    .filter(el => el.length !== 0)
```

Comment:
looks too complex, you can juse split elements in that array by ":" and set the firstElement.trim() as the key and the secondElement.trim() as the value to the object